### PR TITLE
refactor: drop named groups, they are not supported in Hermes

### PR DESCRIPTION
**Pull request by andrejborstnik**
We need this for RN64, at least until Hermes adds named regex groups support. This is the only named regex group we use in the mobile app

### CODE REVIEW

File: src/ManagedNodeAddress.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * @typedef {import("./address_book/NodeAddress.js").default} NodeAddress
  */
 
-const HOST_AND_PORT = /^(?<address>.*)(:(?<port>\d+))?/;
+const HOST_AND_PORT = /^(.*)(:(\d+))?/;
```
Comment (andrejborstnik) [inline src/ManagedNodeAddress.js:8]: Side-note this regex will not catch the port, `.*` will always consume it. (If this is important to you) we should fix it @feri42

Comment (feri42) [reply src/ManagedNodeAddress.js:8]: Looks to me like the original RE never catches to port either.

Comment (andrejborstnik) [reply src/ManagedNodeAddress.js:8]: Yes, that is my point. This PR is a pure refactor that removes named groups for compatibility. But while testing I noticed the above & we can fix it separately if needed

**Review by feri42 — APPROVED** [review]
utACK. I don't think we need to do anything. The code still mostly works by accident. I have files a bug report upstream: https://github.com/hashgraph/hedera-sdk-js/issues/793

**Comment by ChALkeR** [comment]
This fails to compile. I'll post an update shortly.

**Comment by ChALkeR** [comment]
Fixed in f8dec48e92ffbba2364fd51782bfe05e12951a2c.
Note how this doesn't reassign the variable to a different type anymore.

File: src/ManagedNodeAddress.js [hunk 2]
```diff
@@ -16,25 +16,22 @@ export default class ManagedNodeAddress {
      */
     constructor(props = {}) {
         if (props.address != null) {
-            const hostAndPortResult = HOST_AND_PORT.exec(props.address);
+            let hostAndPortResult = HOST_AND_PORT.exec(props.address);
```
Comment (ChALkeR) [inline src/ManagedNodeAddress.js:19]: Should have been `const` and a different variable instead to avoid reassigning.
This failed type checker.

**Pull request merged**